projects/igniteui-angular/src/lib/directives/date-time-editor/date-time-editor.directive.ts

```diff
@@ -0,0 +1,531 @@
+   *  for the editor.
+   * @example
+   *```html
+   * <input [igxDateTimeEditor]="'dd/MM/yyyy'">
+   *```
+   */
+  @Input(`igxDateTimeEditor`)
+  public set inputFormat(value: string) {
+    if (value) {
+      this._format = value;
+    }
+    const mask = this.buildMask(this.inputFormat);
+    this.mask = value.indexOf('tt') !== -1 ? mask.substring(0, mask.length - 2) + 'LL' : mask;
+  }
+
+  public get inputFormat(): string {
+    return this._format;
+  }
+
+  /**
+   * An @Input property that gets/sets the component date value.
+   * @example
+   * ```html
+   * <input igxDateTimeEditor [value]="date">
+   * ```
+   */
+  @Input()
+  public set value(value: Date) {
+    this._value = value;
+    this.updateMask();
+  }
+
+  public get value() {
+    return this._value;
+  }
+
+  /**
+   * Emitted when the editor's value has changed.
+   * @example
+   * ```html
+   * <input igxDateTimeEditor (valueChanged)="onValueChanged($event)"/>
+   * ```
+   */
+  @Output()
+  public valueChanged = new EventEmitter<IgxDateTimeEditorEventArgs>();
+
+  /**
+   * Emitted when the editor is not within a specified range.
+   * @example
+   * ```html
+   * <input igxDateTimeEditor [minValue]="minDate" [maxValue]="maxDate" (validationFailed)="onValidationFailed($event)"/>
+   * ```
+   */
+  @Output()
+  public validationFailed = new EventEmitter<IgxDateTimeEditorEventArgs>();
+
+  private _value: Date;
+  private _document: Document;
+  private _isFocused: boolean;
+  private _format = 'dd/MM/yyyy';
+  private _oldValue: Date | string;
+  private _dateTimeFormatParts: DatePartInfo[];
+  private onTouchCallback = (...args: any[]) => { };
+  private onChangeCallback = (...args: any[]) => { };
+
+  private get literals() {
+    const literals = [];
+    for (const char of this.mask) {
+      if (char.match(/[^0lL]/)) { literals.push(char); }
+    }
+
+    return literals;
+  }
+
+  private get emptyMask() {
+    return this.maskParser.applyMask(this.inputFormat, this.maskOptions);
+  }
+
+  private get targetDatePart(): DatePart {
+    if (this._document.activeElement === this.nativeElement) {
+      return this._dateTimeFormatParts.find(p => p.start <= this.selectionStart && this.selectionStart <= p.end).type;
+    } else {
+      if (this._dateTimeFormatParts.some(p => p.type === DatePart.Date)) {
+        return DatePart.Date;
+      } else if (this._dateTimeFormatParts.some(p => p.type === DatePart.Hours)) {
+        return DatePart.Hours;
+      }
+    }
+  }
+
+  constructor(
+    protected elementRef: ElementRef,
+    protected maskParser: MaskParsingService,
+    protected renderer: Renderer2,
+    @Inject(DOCUMENT) private document: any) {
+    super(elementRef, maskParser, renderer);
+    this._document = this.document as Document;
+  }
+
+  /** @hidden @internal */
+  public ngOnInit(): void {
+    this._dateTimeFormatParts = DatePickerUtil.parseDateTimeFormat(this.inputFormat);
```

Comment:
So if the input format changes after this - date parts remain out of sync? This should be done in `onChanges` or the setter if it does _not_ depend on any other property of the component